ehttp/src/streaming/native.rs

```diff
@@ -0,0 +1,88 @@
+use std::collections::BTreeMap;
+use std::ops::ControlFlow;
+
+use crate::Request;
+
+use super::Part;
+use crate::types::PartialResponse;
+
+pub fn fetch_streaming_blocking(
+    request: Request,
+    on_data: Box<dyn Fn(crate::Result<Part>) -> ControlFlow<()> + Send>,
+) {
+    let mut req = ureq::request(&request.method, &request.url);
+
+    for header in &request.headers {
+        req = req.set(header.0, header.1);
+    }
+
+    let resp = if request.body.is_empty() {
+        req.call()
+    } else {
+        req.send_bytes(&request.body)
+    };
+
+    let (ok, resp) = match resp {
+        Ok(resp) => (true, resp),
+        Err(ureq::Error::Status(_, resp)) => (false, resp), // Still read the body on e.g. 404
+        Err(ureq::Error::Transport(error)) => {
+            on_data(Err(error.to_string()));
+            return;
+        }
+    };
+
+    let url = resp.get_url().to_owned();
+    let status = resp.status();
+    let status_text = resp.status_text().to_owned();
+    let mut headers = BTreeMap::new();
+    for key in &resp.headers_names() {
+        if let Some(value) = resp.header(key) {
+            // lowercase for easy lookup
+            headers.insert(key.to_ascii_lowercase(), value.to_owned());
+        }
+    }
+
+    let response = PartialResponse {
+        url,
+        ok,
+        status,
+        status_text,
+        headers,
+    };
+    if on_data(Ok(Part::Response(response))).is_break() {
+        return;
+    };
+
+    let mut reader = resp.into_reader();
+    loop {
+        let mut buf = vec![0; 2048];
```

Comment:
Is there a rational for the 2048 size?

MTU is usually around 1500 bytes, but I'm not sure that really matters here.

Reply:
I just rounded up common path MTU to something that seemed reasonable.